File: lib/backend/models/waste_model.js
const mongoose = require('mongoose');

const wasteSchema = mongoose.Schema(
    {
        uid: {
            type: String,
            required: true,
        },
        title: {
            type: String,
            required: true,
        },
        des: {
            type: String,
            required: true,
        },
        amountPerUnit: {
            type: Number,
            required: true,
        },
        numberOfUnits: {
            type: Number,
            required: true,
        },
        total: {
            type: Number,
            required: true,
        },
        urls: [
                {
                    type: String,
                    require: true,
                },
        ],
        address: {
            type: String,
            required: true,
        },
        latlng: {
            latitude: {
                type: Number,
            },
            longitude: {
                type: Number,
            },
        }
    }
);

const schema = mongoose.model('sellSchema', wasteSchema);

module.exports = schema;

File: lib/models/waste_model.dart
import 'package:flutter/material.dart';
import 'package:latlong2/latlong.dart';

class WasteModel {
  String title, des, uid, address;
  double amountPerUnit, total;
  int numberOfUnits;
  List<dynamic> urls;
  LatLng? latlng;

  WasteModel({
    required this.title,
    required this.uid,
    required this.amountPerUnit,
    required this.des,
    required this.numberOfUnits,
    required this.total,
    required this.urls,
    required this.address,
    this.latlng,
  });
}
